doc/source/notebooks/monitor-tensorboard.ipynb

```diff
@@ -88,425 +53,423 @@
-   },
+   "metadata": {},
    "outputs": [],
    "source": [
-    "global_step = tf.Variable(0, trainable=False, name=\"global_step\")\n",
-    "m.enquire_session().run(global_step.initializer)"
+    "session = m.enquire_session()\n",
+    "global_step = mon.create_global_step(session)"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Next, we create the optimiser action. `make_optimize_action` also creates the optimisation tensor, which is added to the computational graph. Later, the saver will store the whole graph, and so can also restore the exact optimiser state."
+    "## Construct the monitor\n",
+    "\n",
+    "Next we need to construct the monitor. `gpflow.training.monitor` provides classes that are building blocks for the monitor. Essengially, a monitor is a function that is provided as a callback to an optimiser. It consists of a number of tasks that may be executed at each step, subject to their running condition.\n",
+    "\n",
+    "In this example, we want to:\n",
+    "- log certain scalar parameters in TensorBoard,\n",
+    "- log the full optimisation objective (log marginal likelihood bound) periodically, even though we optimise with minibatches,\n",
+    "- store a backup of the optimisation process periodically,\n",
+    "- log performance for a test set periodically.\n",
+    "\n",
+    "We will define these tasks as follows:"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": 5,
-   "metadata": {
-    "collapsed": true
-   },
-   "outputs": [],
-   "source": [
-    "adam = gpflow.train.AdamOptimizer(0.01).make_optimize_action(m, global_step=global_step)"
-   ]
-  },
-  {
-   "cell_type": "markdown",
    "metadata": {},
-   "source": [
-    "## Creating actions for keeping track of the optimisation\n",
-    "We now create an instance of `FileWriter`, which will save the TensorBoard logs to a file. This object needs to be shared between all `gpflow_monitor.TensorBoard` objects, if they are to write to the same path."
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 6,
-   "metadata": {
-    "collapsed": true
-   },
    "outputs": [],
    "source": [
-    "# create a filewriter for summaries\n",
-    "fw = tf.summary.FileWriter('./model_tensorboard', m.graph)"
+    "print_task = mon.PrintTimingsTask().with_name('print')\\\n",
+    "    .with_condition(mon.PeriodicIterationCondition(10))\\\n",
+    "    .with_exit_condition(True)\n",
+    "\n",
+    "sleep_task = mon.SleepTask(0.01).with_name('sleep').with_name('sleep')\n",
+    "\n",
+    "saver_task = mon.CheckpointTask('./monitor-saves').with_name('saver')\\\n",
+    "    .with_condition(mon.PeriodicIterationCondition(10))\\\n",
+    "    .with_exit_condition(True)\n",
+    "\n",
+    "std_tboard_task = mon.StandardTensorBoardTask(m, './model-tensorboard').with_name('std_tboard')\\\n",
+    "    .with_condition(mon.PeriodicIterationCondition(10))\\\n",
+    "    .with_exit_condition(True)\n",
```

Comment:
No, we can scrap with_global_step_tensor. I added it because I thought we need global_step it in the summary. Since we don't actually need it there it makes things easier.